pipenv/cli.py

```diff
@@ -2056,6 +2059,9 @@ def lock(three=None, python=False, verbose=False, requirements=False, dev=False,
 
 def do_shell(three=None, python=False, fancy=False, shell_args=None):
 
+    # Ensure that virtualenv is available.
+    ensure_project(three=three, python=python, validate=False)
+
```

Comment:
Doesn't this still always create a virtualenv though?

Reply:
This is the state the codebase was in prior to #324. Kenneth added this 3 months ago, so I don't think we should completely remove it unless we know why we're doing it.

Reply:
I agree and this looks sensible to me but I think it will still cause the issue which caused him to comment it out in the first place, or if it doesn't I'm not seeing why not

Reply:
I'm not entirely sure it doesn't. However, that line has been in either `shell` or `do_shell` since 7.0.0 and I'm hesitant to remove it without understanding why it was there and we're now suddenly having an issue with it.

We should probably file a follow up issue with the specific case and address that separately though.